src/processTargets/modifiers/ContainingScopeStage.ts

```diff
@@ -93,3 +112,71 @@ export class ContainingScopeStage implements ModifierStage {
     return [constructScopeRangeTarget(isReversed, startScope, endScope)];
   }
 }
+
+function expandFromPosition(
+  scopeHandler: ScopeHandler,
+  editor: TextEditor,
+  position: Position,
+  direction: Direction,
+  ancestorIndex: number,
+): TargetScope | undefined {
+  let nextAncestorIndex = 0;
+  for (const scope of scopeHandler.generateScopes(editor, position, direction, {
+    containment: "required",
+  })) {
+    if (nextAncestorIndex === ancestorIndex) {
+      return scope;
+    }
+
+    // Because containment is required, and we are moving in a consistent
+    // direction (ie forward or backward), each scope will be progressively
+    // larger
+    nextAncestorIndex += 1;
+  }
+
+  return undefined;
+}
+
+function getPreferredScopeTouchingPosition(
+  scopeHandler: ScopeHandler,
+  editor: TextEditor,
+  position: Position,
+): TargetScope | undefined {
+  const forwardScope = getContainingScope(
+    scopeHandler,
+    editor,
+    position,
+    "forward",
+  );
+
+  if (forwardScope == null) {
+    return getContainingScope(scopeHandler, editor, position, "backward");
+  }
+
+  if (
+    scopeHandler.isPreferredOver == null ||
+    forwardScope.domain.start.isBefore(position)
+  ) {
+    return forwardScope;
+  }
+
+  const backwardScope = getContainingScope(
+    scopeHandler,
+    editor,
+    position,
+    "backward",
+  );
+
+  // If there is no backward scope, or if the backward scope is an ancestor of
+  // forward scope, return forward scope
+  if (
+    backwardScope == null ||
+    backwardScope.domain.contains(forwardScope.domain)
+  ) {
+    return forwardScope;
+  }
+
+  return scopeHandler.isPreferredOver(backwardScope, forwardScope) ?? false
```

Comment:
`isPreferredOver` can't be null here. You shouldn't need the nullish operator.

Reply:
Yes it can.  See implementation of token preferences in TokenScopeHandler

Reply:
Ok so it can return undefined. Gotcha